nil/go-ibft/core/ibft.go

```diff
@@ -101,14 +105,34 @@ type IBFT struct {
 
 	// validatorManager keeps quorumSize and voting power information
 	validatorManager *ValidatorManager
+
+	// metricAttrs is the attributes to be used for metrics
+	metricAttrs []attribute.KeyValue
+
+	// roundHistogram is the histogram for round duration
+	roundHistogram metric.Float64Histogram
+
+	// sequenceHistogram is the histogram for sequence duration
+	sequenceHistogram metric.Float64Histogram
 }
 
 // NewIBFT creates a new instance of the IBFT consensus protocol
 func NewIBFT(
 	log Logger,
 	backend Backend,
 	transport Transport,
-) *IBFT {
+	attrs ...attribute.KeyValue,
+) (*IBFT, error) {
+	roundHistogram, err := meter.Float64Histogram("round.duration", metric.WithDescription("Round duration"))
+	if err != nil {
+		return nil, err
+	}
+
+	sequenceHistogram, err := meter.Float64Histogram("sequence.duration", metric.WithDescription("Sequence duration"))
+	if err != nil {
+		return nil, err
+	}
+
 	return &IBFT{
```

Comment:
Maybe we could make a `NewIBFTWithMetrics` wrapper and use it in the combat code and leave it as is in the tests?

Reply:
Ok. Maybe it's better. Applied.